### ISSUE TITLE
Debug-mode crash during table fuzzing: server_name_map doesn't contain the expected server

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#4973]
I was testing the new `--kill` option for `table_fuzzer.py` against `v2.1.x`, and ran into this crash with the command line `table_fuzzer.py --servers 7 --threads 48 --changefeeds --kill`

```
error: Error in ../src/clustering/administration/servers/server_metadata.cc at line 37:
error: server_name_map doesn't contain the expected server. expected: 4ef020da-4760-4436-b024-38d1145ae41f found: 60893a92-8864-4df4-afa2-c174b606ffe8=g, b338d283-f296-4713-88f1-d81246f6c63a=e, bb7f350a-c712-4da3-931f-f6b0785ea2df=a, bd816db7-dd3d-42af-ac12-fb0a055daae9=c, e22b87e7-60e5-47e8-bc5b-17688173e951=b
error: Backtrace:
2015-10-20T17:07:05.992028597 22.199573s error: Tue Oct 20 17:07:04 2015
1 [0x1bb88ac]: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:101
2 [0x17dd771]: backtrace_t::backtrace_t() at backtrace.cc:203
3 [0x17ddddf]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
4 [0x17dd6bf]: format_backtrace(bool) at backtrace.cc:198
5 [0x17d15a2]: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
6 [0x1d83eda]: server_name_map_t::get(uuid_u const&) const at server_metadata.cc:36
7 [0x1bda5f7]: table_availability_issue_t::build_info_and_description(cluster_semilattice_metadata_t const&, server_config_client_t*, table_meta_client_t*, admin_identifier_format_t, ql::datum_t*, datum_string_t*) const at table.cc:90
8 [0x1bbb30e]: issue_t::to_datum(cluster_semilattice_metadata_t const&, server_config_client_t*, table_meta_client_t*, admin_identifier_format_t, ql::datum_t*) const at issue.cc:17
9 [0x1bbbf76]: issues_artificial_table_backend_t::read_all_rows_as_vector(signal_t*, std::vector<ql::datum_t, std::allocator<ql::datum_t> >*, admin_err_t*) at issues_backend.cc:55
10 [0x1b020e2]: artificial_table_backend_t::read_all_rows_as_stream(ql::backtrace_id_t, ql::datumspec_t const&, sorting_t, signal_t*, counted_t<ql::datum_stream_t>*, admin_err_t*) at backend.cc:17
11 [0x1afef88]: artificial_table_t::read_all(ql::env_t*, std::string const&, ql::backtrace_id_t, std::string const&, ql::datumspec_t const&, sorting_t, read_mode_t) at artificial_table.cc:74
12 [0x1ae3b67]: ql::table_t::as_seq(ql::env_t*, std::string const&, ql::backtrace_id_t, ql::datum_range_t const&, sorting_t) at val.cc:224
13 [0x1ae33fc]: ql::table_slice_t::as_seq(ql::env_t*, ql::backtrace_id_t) at val.cc:174
14 [0x1ae5bf8]: ql::val_t::as_seq(ql::env_t*) at val.cc:561
15 [0x1a57656]: ql::changes_term_t::eval_impl(ql::scope_env_t*, ql::args_t*, ql::eval_flags_t) const at seq.cc:416
16 [0x19d9b0f]: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:225
17 [0x1a9fe3e]: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:353
18 [0x1bb20cd]: ql::query_cache_t::ref_t::run(ql::env_t*, Response*) at query_cache.cc:311
19 [0x1bb1894]: ql::query_cache_t::ref_t::fill_response(Response*) at query_cache.cc:249
20 [0x1a9f692]: ql::run(ql::query_id_t&&, ql::protob_t<Query>, Response*, ql::query_cache_t*, signal_t*) at term.cc:269
21 [0x1baee4a]: rdb_query_server_t::run_query(ql::query_id_t&&, ql::protob_t<Query> const&, Response*, ql::query_cache_t*, signal_t*) at query_server.cc:49
```

**Comment by Tryneus** [comment]
Note that in release mode, this will just print an error and continue on with no problem - the table availability issue will just have an `__unknown_server__` value in it.

**Comment by Tryneus** [comment]
Fix is up in review 3293.

**Comment by Tryneus** [comment]
Fix was approved and merged to `next` and `v2.1.x` in abf9634 and dfaffe7, respectively.  Will be in release 2.1.6.

**Comment by Tryneus** [comment]
Just ran into this again on `next` with a slightly different backtrace, so it isn't completely fixed:

```
Version: rethinkdb 2.1.0-0BETA2-375-gabf963-dirty (debug) (GCC 4.6.3)
error: Error in ../src/clustering/administration/servers/server_metadata.cc at line 37:
error: server_name_map doesn't contain the expected server. expected: 6ac0bce2-2060-4613-912f-fc7446588df9 found: 91effbfd-d8c4-4c10-917f-d7b88a292702=c, ccf229af-fd3d-4c92-8596-68fa96bbd40b=d
error: Backtrace:
error: Fri Oct 23 11:52:04 2015

       1 [0x1d22d28]: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:101
       2 [0x190af01]: backtrace_t::backtrace_t() at backtrace.cc:203
       3 [0x190b56f]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
       4 [0x190ae4f]: format_backtrace(bool) at backtrace.cc:198
       5 [0x18fed0e]: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       6 [0x1f08aea]: server_name_map_t::get(uuid_u const&) const at server_metadata.cc:36
       7 [0x1e86963]: convert_raft_leader_to_datum(table_status_t const&, admin_identifier_format_t) at table_status.cc:97
       8 [0x1e86cef]: convert_table_status_to_datum(table_status_t const&, admin_identifier_format_t) at table_status.cc:134
       9 [0x1e87155]: table_status_artificial_table_backend_t::format_row(uuid_u const&, table_config_and_shards_t const&, ql::datum_t const&, signal_t*, ql::datum_t*) at table_status.cc:162
       10 [0x1e78393]: common_table_artificial_table_backend_t::read_all_rows_as_vector(signal_t*, std::vector<ql::datum_t, std::allocator<ql::datum_t> >*, admin_err_t*) at table_common.cc:48
       11 [0x1c77956]: artificial_table_backend_t::read_all_rows_as_stream(ql::backtrace_id_t, ql::datumspec_t const&, sorting_t, signal_t*, counted_t<ql::datum_stream_t>*, admin_err_t*) at backend.cc:17
       12 [0x1c747fc]: artificial_table_t::read_all(ql::env_t*, std::string const&, ql::backtrace_id_t, std::string const&, ql::datumspec_t const&, sorting_t, read_mode_t) at artificial_table.cc:74
       13 [0x1cf925f]: ql::table_t::as_seq(ql::env_t*, std::string const&, ql::backtrace_id_t, ql::datum_range_t const&, sorting_t) at val.cc:224
       14 [0x1cf8af4]: ql::table_slice_t::as_seq(ql::env_t*, ql::backtrace_id_t) at val.cc:174
       15 [0x1cfb2f0]: ql::val_t::as_seq(ql::env_t*) at val.cc:561
       16 [0x1b74448]: ql::changes_term_t::eval_impl(ql::scope_env_t*, ql::args_t*, ql::eval_flags_t) const at seq.cc:419
       17 [0x1bfdf23]: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:220
       18 [0x1960d30]: ql::runtime_term_t::eval_on_current_stack(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:273
       19 [0x1960eaa]: /home/ssd2/grey/rethinkdb/build/debug/rethinkdb() [0x1960eaa] at 0x1960eaa ()
       20 [0x1961159]: /home/ssd2/grey/rethinkdb/build/debug/rethinkdb() [0x1961159] at 0x1961159 ()
       21 [0x1960ef7]: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:294
       22 [0x1bd7c11]: ql::query_cache_t::ref_t::run(ql::env_t*, ql::response_t*) at query_cache.cc:261
       23 [0x1bd73d8]: ql::query_cache_t::ref_t::fill_response(ql::response_t*) at query_cache.cc:198
       24 [0x19b0417]: rdb_query_server_t::run_query(ql::query_params_t*, ql::response_t*, signal_t*) at query_server.cc:39
       25 [0x1919354]: /home/ssd2/grey/rethinkdb/build/debug/rethinkdb() [0x1919354] at 0x1919354 ()
       26 [0x191945d]: /home/ssd2/grey/rethinkdb/build/debug/rethinkdb() [0x191945d] at 0x191945d ()
       27 [0x19190d3]: /home/ssd2/grey/rethinkdb/build/debug/rethinkdb() [0x19190d3] at 0x19190d3 ()
       28 [0x191ed82]: callable_action_instance_t<void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}>::run_action() at callable_action.hpp:31
       29 [0x172c268]: callable_action_wrapper_t::run() at runtime_utils.cc:43
       30 [0x173446f]: coro_t::run() at coroutines.cc:207
```

**Comment by VeXocide** [comment]
Sorry, I forgot to update this issue. A fix was merged to `next` via commit b4a41752c8a32bd2e971a5381aa038a2f09a6371 after a review by @Tryneus.
